project1-sps/script.js

```diff
@@ -1,4 +1,79 @@
+var NumPlayerWins = 0;
+var NumPlayerlosses = 0;
+var NumPlayerdraws = 0;
+
+var mode = "get user name";
+var UserName = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue;
+  if (mode == "get user name") {
+    //save username in global variable
+    UserName = input;
+    // change mode to play game
+    mode = "play game";
+    myOutputValue = `Hello ${UserName}`;
+  } else {
+    var playerInput = input;
+    var computerInput = getRandomObject();
+    console.log(computerInput);
+
+    var allWinningConditions =
+      (playerInput == SCISSORS && computerInput == PAPER) ||
+      (playerInput == STONE && computerInput == SCISSORS) ||
+      (playerInput == PAPER && computerInput == STONE);
+
+    var allLosingConditions =
+      (computerInput == SCISSORS && playerInput == PAPER) ||
+      (computerInput == STONE && playerInput == SCISSORS) ||
+      (computerInput == PAPER && playerInput == STONE);
+
+    var myOutputValue = "";
+
+    if (allWinningConditions) {
+      NumPlayerWins += 1;
+      myOutputValue = `Hello ${UserName}. <br> you win! <br> Total Wins ${NumPlayerWins}. Total Losses ${NumPlayerlosses}. Total Draws ${NumPlayerdraws}`;
+    } else if (allLosingConditions) {
+      NumPlayerlosses += 1;
+      myOutputValue = `Hello ${UserName}.  <br> you lose! <br>Total Wins ${NumPlayerWins}. Total Losses ${NumPlayerlosses}. Total Draws ${NumPlayerdraws}`;
+    } else {
+      NumPlayerdraws += 1;
+      myOutputValue = `Hello ${UserName}.  <br> draw. <br>Total Score ${NumPlayerWins}. Total Losses ${NumPlayerlosses}. Total Draws ${NumPlayerdraws}`;
```

Comment:
Nice template literals here! Since they are all outputting the same message - we can store the result (you win/you lose/draw) in another variable and do `myOutputValue = "....."`  just once after all the if conditions.